### ISSUE TITLE
Need Help

### ISSUE THREAD

**Issue by MohammedAliSharafuddin** [klarsen1/bstspost#3]
I am using a "10 years" monthly time-series data to run this code. But getting "Error in data.frame(c(10^as.numeric(-colMeans(bsts.model$one.step.prediction.errors[-(1:burn),  : 
  arguments imply differing number of rows: 144, 120"

d2 <- data.frame(

    # fitted values and predictions

    c(10^as.numeric(-colMeans(bsts.model$one.step.prediction.errors[-(1:burn),])+y),  

    10^as.numeric(p$mean)),

    # actual data and dates 

    as.numeric(AirPassengers),

    as.Date(time(AirPassengers)))

names(d2) <- c("Fitted", "Actual", "Date")

Please help me to sort out this error.

**Comment by klarsen1** [comment]
you probably have holes in your input time series.

K

**Comment by mcui123** [comment]
I have actually encountered same issue as @MohamedaliS and couldn't figure out why. 

@klarsen1  Can you specify what is "have holes" in time series? 
For example, for my daily time-series data, does that mean I don't have data for each single day? Do I have to have a data entry for each day? What if I have multiple data entries in a day? 

Also, the fact that the traceback error suggests its off by exactly the "horizon = xx" (in my case) makes me wonder what went wrong.... 

Thanks so much in advance!

**Comment by klarsen1** [comment]
Yes. This typically happens if you don't have data for every period (day,
week, month, etc). This creates all sorts of havoc when calculating model
accuracy, etc.

Also, when running time series models it's generally a bad idea to skip
periods – whether it's STL or ARIMA.

Also, if you data is daily, you cannot have two or more observations for a
single day. Then your data is not daily.

Kim

**Comment by mcui123** [comment]
Thanks so much @klarsen1 ! I was able to fix the issue from your recommendation. 

I have tried your code on my own set of data, and have encountered another issue - the burn out rate is very high (e.g. with 500 MCMC iterations, bsts suggested a burn out number of 497), which I think could be the reason why my MAPE is as high as 70%+. 

FYI. I have 3 years of daily data (I aggregated by day so that I don't have multiple entries in a same day). My holdout period is 30 days. Just wondering if you have any insights as to whether there are any other way to improve model accuracy? 

Best, 
JC

**Comment by klarsen1** [comment]
I’d say play with the level term standard error. It’s described in the
vignette.

K
